specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/stable/2017-04-01/azureactivedirectory.json

```diff
@@ -0,0 +1,507 @@
+          "type": "string"
+        },
+        "description": {
+          "description": "The description. The localized friendly description for the operation, as it should be shown to the user. It should be thorough, yet concise – it will be used in tool tips and detailed views. Prescriptive guidance for namespaces: Read any 'display.provider' resource Create or Update any 'display.provider' resource Delete any 'display.provider' resource Perform any other action on any 'display.provider' resource Prescriptive guidance for namespaces: Read any 'display.resource' Create or Update any 'display.resource' Delete any 'display.resource' 'ActionName' any 'display.resources'",
+          "type": "string"
+        }
+      }
+    },
+    "OperationsDiscovery": {
+      "description": "Operations discovery class.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the API. The name of the operation being performed on this particular object. It should match the action name that appears in RBAC / the event service. Examples of operations include: * Microsoft.Compute/virtualMachine/capture/action * Microsoft.Compute/virtualMachine/restart/action * Microsoft.Compute/virtualMachine/write * Microsoft.Compute/virtualMachine/read * Microsoft.Compute/virtualMachine/delete Each action should include, in order: (1) Resource Provider Namespace (2) Type hierarchy for which the action applies (e.g. server/databases for a SQL Azure database) (3) Read, Write, Action or Delete indicating which type applies. If it is a PUT/PATCH on a collection or named value, Write should be used. If it is a GET, Read should be used. If it is a DELETE, Delete should be used. If it is a POST, Action should be used.",
+          "type": "string"
+        },
+        "display": {
+          "$ref": "#/definitions/Display",
+          "description": "Object type"
+        },
+        "origin": {
+          "description": "Origin. The intended executor of the operation; governs the display of the operation in the RBAC UX and the audit logs UX. Default value is \"user,system\"",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/OperationsDiscoveryProperties",
+          "description": "Properties. Reserved for future use."
+        }
+      }
+    },
+    "OperationsDiscoveryCollection": {
+      "description": "Collection of ClientDiscovery details.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "The ClientDiscovery details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OperationsDiscovery"
+          }
+        }
+      }
+    },
+    "DiagnosticSettingsCategory": {
+      "properties": {
+        "categoryType": {
+          "enum": ["Logs"],
+          "type": "string",
+          "x-ms-enum": {
```

Comment:
Here is recommended to add: 
```
"x-ms-enum": {
  "name": "categoryType",
  "modelAsString": true
}
```
`"modelAsString": true` will avoid breaking changes in the generated types in the future.